### DISCUSSION TITLE
How can I use a string index properly when selecting a prisma model?

### DISCUSSION THREAD

**Discussion by emmanuel-londono** [prisma/prisma#21566] [Q&A] [answered]
### Question

I'd like to use the following code to dynamically search for a primsa model:

```ts
async function paginateBySkipTake(req: Request, res: Response, next: NextFunction) {
    //If skip or take is empty, make default values appropriate.
    const skip = !isNaN(parseInt(req.query.skip as string)) ? parseInt(req.query.skip as string) : 0;
    const take = !isNaN(parseInt(req.query.take as string)) ? parseInt(req.query.take as string) : 10;
    const url = req.originalUrl;
    try {
        const modelName = await determineModelName(url, prisma);
        if (modelName) {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            const count = await (prisma as any)[modelName].count({ skip, take });
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            const record = await (prisma as any)[modelName].findMany({ skip, take });
            res.locals.record = { "Total Count": count, ...record };
        } else {
            res.locals.record = { error: "Specific record ID and field name is required." };
        }
    } catch (err) {
        catchHandler(res, err);
    } finally {
        next();
        await prisma.$disconnect();
    }
}
```

Is there any way to type cast the prisma object properly, to prevent using any? Without using any, I'm not allowed to use string indexes.

### Information about Prisma Schema, Client Queries and Environment (optional)

OS: Windows
Database: PostgreSQL
Node.js version: v16.14.2
RestAPI: Expressjs
Language: Typescript
prisma version: 5.3.1
@prisma/client version: 5.4.2

**Comment by ludralph** [comment, marked as answer]
Hi @emmanuel-londono 👋 

Thank you for raising this question. 

You can do something like this:
```js
const modelType: PrismaClient[typeof modelName] = prisma[modelName];
const count = await modelType.count();
const record = await modelType.findMany();
```
This will tell TypeScript that the `modelType` variable can be any of the Prisma client models, depending on the value of the `modelName` variable.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by LePtiDev** [reply]
Hello, i don't understand how do you fix it ? I got an error on the find many like this.

<img width="708" alt="Capture d’écran 2023-11-24 à 11 30 39" src="https://github.com/prisma/prisma/assets/47982394/38ea5c51-343e-4ddd-b0fa-fffe0a2333ea">
<img width="1177" alt="Capture d’écran 2023-11-24 à 11 31 14" src="https://github.com/prisma/prisma/assets/47982394/3a57a67d-e09b-4906-8a72-46ef3a7c9795">

**Reply by Chrys-code** [reply]
Facing the same error you do @LePtiDev.
